test/components/util/util.go

```diff
@@ -176,13 +176,24 @@ func WaitForDeployment(t *testing.T, client kubernetes.Interface, ns, name strin
 	}
 
 	// Check the readiness of the pods
-	labelSet := labels.Set(deploy.Labels)
+	labelMap, err := metav1.LabelSelectorAsMap(deploy.Spec.Selector)
+	if err != nil {
+		t.Fatalf("converting label selector to map: %v", err)
+	}
+
 	if err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
-		pods, err := client.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSet.String()})
+		listOptions := metav1.ListOptions{LabelSelector: labels.SelectorFromSet(labelMap).String()}
```

Comment:
This call to `labels.SelectorFromSet` can be avoided by converting `deploy.Spec.Selector` to type `Selector` instead of map using funciton `LabelSelectorAsSelector`. Once you have `Selector` then you can call `String()` on it.

Reply:
Ah, good to know, will change. I didn't put much thought on it, was just applying function until the type matched :smile:

Reply:
Done.